insecure/corruptible/network.go

```diff
@@ -0,0 +1,370 @@
+package corruptible
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"sync"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/ipfs/go-datastore"
+	"github.com/libp2p/go-libp2p-core/protocol"
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc"
+
+	"github.com/onflow/flow-go/crypto/hash"
+	"github.com/onflow/flow-go/engine/execution/ingestion"
+	"github.com/onflow/flow-go/engine/execution/state/delta"
+	"github.com/onflow/flow-go/engine/execution/utils"
+	verutils "github.com/onflow/flow-go/engine/verification/utils"
+	"github.com/onflow/flow-go/engine/verification/verifier"
+	"github.com/onflow/flow-go/insecure"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	flownet "github.com/onflow/flow-go/network"
+	"github.com/onflow/flow-go/network/channels"
+	"github.com/onflow/flow-go/utils/logging"
+)
+
+// Network is a wrapper around the original flow network, that allows a remote attacker
+// to take control over its ingress and egress traffic flows.
+// A remote attacker can register itself to this corruptible network.
+// Whenever any corruptible conduit receives an event from its engine, it relays the event to this
+// network, which in turn is relayed to the register attacker.
+// The attacker can asynchronously dictate to the network to send messages on behalf of the node.
+type Network struct {
```

Comment:
we have `insecure/corruptible/network.go` and `insecure/attacknetwork/attackNetwork.go` - it's confusing to have 2 corruptible / insecure networks. Can we rename `insecure/attacknetwork/attackNetwork.go` so it's less confusing?

Reply:
I think they are self-explanatory in the terms of interfaces they implement; the `Network` implements the `flownetwork.Network` (as we explicitly added `var _ flownet.Network = &Network{}`) which is the network layer of the node (godoc). On the other hand `AttackNetwork` implements `insecure.AttackNetwork`, which is the networking interface available to the attacker (godoc). We have several similar naming pairs across our codebase as well, and I think as long as we set the interface and godoc clear, it is easy to track them. Nevertheless, if you have any better naming suggestion please let me know, and we can consider renaming them. 

P.S: following your comment, I added `var _ insecure.AttackNetwork = &AttackNetwork{}` explicit interface conversion for attack network in https://github.com/onflow/flow-go/pull/2807/commits/fd0aa7a8e82b46bf4751eb7dddcbc336572d051b.